pkg/reconciler/pingsource/resources/mt_receive_adapter.go

```diff
@@ -50,6 +54,16 @@ func MakeMTReceiveAdapter(args MTArgs) *v1.Deployment {
 		ObjectMeta: metav1.ObjectMeta{
 			Namespace: system.Namespace(),
 			Name:      args.MTAdapterName,
+			OwnerReferences: []metav1.OwnerReference{
+				{
+					APIVersion:         v1.SchemeGroupVersion.String(),
+					Kind:               "Deployment",
+					Name:               os.Getenv("CONTROLLER_NAME"),           // guarantee to be non-empty
```

Comment:
you have already pulled this out of the environment, you should pass that var down so it is not also pulled from the env here.